adders/drizzle/config/tests.js

```diff
@@ -0,0 +1,100 @@
+        { ...defaultOptionValues, database: "postgresql", postgresql: "postgres.js", docker: true },
+    ],
+    runSynchronously: true,
+    files: [
+        {
+            name: ({ kit }) => `${kit.routesDirectory}/+page.svelte`,
+            contentType: "svelte",
+            condition: ({ kit }) => kit.installed,
+            content: ({ html, js }) => {
+                js.common.addFromString(js.ast, "export let data;");
+                html.addFromRawHtml(
+                    html.ast.childNodes,
+                    `
+                    {#each data.users as user}
+                        <span data-test-id="user-id-{user.id}">{user.id} {user.name}</span>
+                    {/each}
+                    `,
+                );
+            },
+        },
+        {
+            name: ({ kit, typescript }) => `${kit.routesDirectory}/+page.server.${typescript.installed ? "ts" : "js"}`,
+            contentType: "script",
+            condition: ({ kit }) => kit.installed,
+            content: ({ ast, common, typescript }) => {
+                common.addFromString(
+                    ast,
+                    `
+                    import { db } from '$lib/server/db';
+                    import { user } from '$lib/server/db/schema.js';
+
+                    export const load = async () => {
+                        await insertUser({ name: 'Foobar', id: 0, age: 20 }).catch((err) => console.error(err));
+
+                        const users = await db.select().from(user);
+
+                        return { users };
+                    };
+
+                    function insertUser(${typescript.installed ? "value: typeof user.$inferInsert" : "value"}) {
+                        return db.insert(user).values(value);
+                    }
+                    `,
+                );
+            },
+        },
+        {
+            // override the config so we can remove strict mode
+            name: ({ typescript }) => `drizzle.config.${typescript.installed ? "ts" : "js"}`,
+            contentType: "text",
+            condition: ({ kit }) => kit.installed,
+            content: ({ typescript, options }) => {
+                return `
+                import { defineConfig } from "drizzle-kit";
+
+                export default defineConfig({
+                    schema: './src/lib/server/db/schema.${typescript.installed ? "ts" : "js"}',
+                    dialect: "${options.database}",
+                    dbCredentials: {
+                        url: process.env.DATABASE_URL
+                    },
+                    verbose: true
+                });
+                `;
+            },
+        },
+        {
+            name: () => "package.json",
+            contentType: "json",
+            content: ({ data }) => {
+                // executes after pnpm install
+                data.scripts["postinstall"] ??= "pnpm run db:push";
+            },
```

Comment:
I would consider this more of a hack than a solution. What do you think?

Wouldn't it be better if we introduce pre and post test hooks?

Reply:
Indeed, it is a hack 😅 

Pre and post test hooks would be great in these cases

Reply:
Let's leave it like this for now, I have opened #389 for tracking